src/Datadog.Trace/Span.cs

```diff
@@ -324,14 +331,34 @@ public void SetException(Exception exception)
         /// <param name="key">The tag's key</param>
         /// <returns> The value for the tag with the key specified, or null if the tag does not exist</returns>
         public string GetTag(string key)
+        {
+            return GetTagValue(key);
+        }
+
+        /// <summary>
+        /// Gets the value (or default/null if the key is not a valid tag) of a tag with the key value passed
+        /// </summary>
+        /// <param name="key">The tag's key</param>
+        /// <returns> The value for the tag with the key specified, or null if the tag does not exist</returns>
+        public TagValue? GetTagValue(string key)
```

Comment:
We should seek to keep the new changes from modifying the public API, and I think we can still do this while enabling the ability for input strings with numerical values to be represented as numbers. Since the original `string GetTag(string)` returns a string, if we ever need to return a number, we can just allocate a new string with that numerical value

Reply:
On second thought, this very well may warrant an addition to the public API. As Greg suggested, we should probably get on a call to figure this out